homeassistant/components/blebox/light.py

```diff
@@ -166,7 +165,7 @@ async def async_turn_on(self, **kwargs):
         else:
             try:
                 await self._feature.async_on(value)
-            except BadOnValueError as ex:
+            except ValueError as ex:
                 _LOGGER.error(
```

Comment:
We prefer raising an exception instead of logging in service handlers. We normally raise `ValueError` on bad user input with a message as exception argument. The message will show in the frontend.

Reply:
Will it be ok if blebox_uniapi raises an exception like this
```
   raise ValueError(
                f"turn_on called with bad parameter ({value} is {type(value)}, compared to {self._off_value} which is "
                f"{type(self._off_value)})"
            )
```
or does the component need to catch this error and reraise with a different message?

Reply:
It's important that the message is clear to the user. Maybe the user doesn't know about all the details of the library message?

Reply:
So maybe I'll go back to the message from before? 
```
            except ValueError:
                raise ValueError(
                    f"Turning on '{self.name}' failed: Bad value {value}"
                )
```